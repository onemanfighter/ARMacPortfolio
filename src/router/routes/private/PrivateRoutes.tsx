import {
  Navigate,
  Route,
  createBrowserRouter,
  createRoutesFromChildren,
} from 'react-router-dom';
import { LazyHomeComponent, LazyMacComponent } from '../../lazyRouting';

export const privateRouter = createBrowserRouter(
  createRoutesFromChildren(
    <Route path="/" element={<LazyMacComponent />}>
      <Route path="" element={<Navigate to="home" replace />} />
      <Route path="home" element={<LazyHomeComponent />} />
      <Route path="*" element={<Navigate to="/home" replace />} />
    </Route>,
  ),
);
