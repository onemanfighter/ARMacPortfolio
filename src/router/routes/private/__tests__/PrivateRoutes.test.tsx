import { render, renderHook } from '@testing-library/react';
import { privateRouter } from '../PrivateRoutes';
import { BrowserRouter, RouterProvider } from 'react-router-dom';

describe('PrivateRoutes', () => {
  it('should render the PrivateRoutes component', () => {
    const { result } = renderHook(() => privateRouter);

    expect(result.current.routes[0].children).toMatchSnapshot();
  });
});
