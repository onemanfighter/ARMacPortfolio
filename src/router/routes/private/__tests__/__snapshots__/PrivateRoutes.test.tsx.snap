// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrivateRoutes should render the PrivateRoutes component 1`] = `
[
  {
    "Component": undefined,
    "ErrorBoundary": undefined,
    "action": undefined,
    "caseSensitive": undefined,
    "children": undefined,
    "element": <LazyHomeComponent />,
    "errorElement": undefined,
    "handle": undefined,
    "hasErrorBoundary": false,
    "id": "0-0",
    "index": undefined,
    "lazy": undefined,
    "loader": undefined,
    "path": "",
    "shouldRevalidate": undefined,
  },
  {
    "Component": undefined,
    "ErrorBoundary": undefined,
    "action": undefined,
    "caseSensitive": undefined,
    "children": undefined,
    "element": <Navigate
      replace={true}
      to="/"
    />,
    "errorElement": undefined,
    "handle": undefined,
    "hasErrorBoundary": false,
    "id": "0-1",
    "index": undefined,
    "lazy": undefined,
    "loader": undefined,
    "path": "*",
    "shouldRevalidate": undefined,
  },
]
`;
